unblob/handlers/archive/_safe_tarfile.py

```diff
@@ -82,7 +82,18 @@ def extract(self, tarinfo: tarfile.TarInfo, extract_root: Path):  # noqa: C901
                     "Absolute path as link target.",
                     "Converted to extraction relative path.",
                 )
-                tarinfo.linkname = f"./{tarinfo.linkname}"
+
+                def calculate_linkname():
+                    root = extract_root.resolve()
+                    path = (extract_root / tarinfo.name).resolve()
+                    common_path = Path(os.path.commonpath([root, path]))
+                    # normally root == common_path
+                    # if it is not, the output will be bad
```

Comment:
We could add an assert here to make sure it is and be noisy if it's not, which is indicative of something wrong with `is_safe_path`.

```python
assert root == common_path
```

Reply:
Yes, this made me thinking of adding an assertion as well.

However an assert would blow up extraction, instead it is explicitly reported now.

A fixup commit is added for review, it needs to be squashed before merge (if accepted).
Converting to draft to prevent accidental merge, but it is supposed to be complete and waiting for review.